### PR TITLE
feat: add `pr-updater` GitHub Action to sync PRs on main push

### PR DESCRIPTION
**Pull request by lemeurherve**
Use https://github.com/maxkomarychev/pr-updater-action

### CODE REVIEW

**Comment by dduportal** [comment]
That a nice addition. I wonder if it excludes by default the draft PRs though? (I would be annoyed to have my Draft PRs being updated all the time)

By the way, I've added 2 new checks rules on this repository to make sure that both Kubernetes Management tasks and Update Check Configuration must be passing in order to auto-merge, to anticipate on the "pr-updated" process

**Comment by lemeurherve** [comment]
>  I wonder if it excludes by default the draft PRs though? 

I don't know, it should be possible to add a condition.

> By the way, I've added 2 new checks rules on this repository to make sure that both Kubernetes Management tasks and Update Check Configuration must be passing in order to auto-merge, to anticipate on the "pr-updated" process

Isn't the `continuous-integration/jenkins/pr-merge` default one not sufficient enough? AFAIU it already regroups and checks both Kubernetes and Updatecli jobs.

File: .github/workflows/pr-updater [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            - main
+
+jobs:
+    autoupdate:
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v1
+        - name: update all pull requests except those with the label 'prevent-pr-updater'
+            uses: maxkomarychev/pr-updater-action@v1.0.0
+            if: !contains(github.event.pull_request.labels.*.name, 'prevent-pr-updater')
```
Comment (lemeurherve) [inline .github/workflows/pr-updater:15]:
```suggestion
            if: !contains(github.event.pull_request.labels.*.name, 'prevent-pr-updater') &&  if: !github.event.pull_request.draft
```

Prevent updating draft pull requests

File: .github/workflows/pr-updater [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    push:
+        branches: 
+            - main
+
+jobs:
+    autoupdate:
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v1
+        - name: update all pull requests except those with the label 'prevent-pr-updater'
```
Comment (lemeurherve) [inline .github/workflows/pr-updater:13]:
```suggestion
        - name: update all pull requests except those with the label 'prevent-pr-updater' and those in draft
```

File: .github/workflows/pr-updater [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+    autoupdate:
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v1
+        - name: update all pull requests except those with the label 'prevent-pr-updater' and those in draft
+            uses: maxkomarychev/pr-updater-action@v1.0.0
+            if: !contains(github.event.pull_request.labels.*.name, 'prevent-pr-updater') && !github.event.pull_request.draft
+            with:
+                token: ${{ secrets.USER_TOKEN }}
```
Comment (lemeurherve) [inline .github/workflows/pr-updater:17]:
```suggestion
                token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by dduportal** [comment]
> Isn't the continuous-integration/jenkins/pr-merge default one not sufficient enough? AFAIU it already regroups and checks both Kubernetes and Updatecli jobs.

Nope: it's sent 2 times (1 time for each build: kube's and updatecli's). 
The last one is the one visible on the UI (and can be failing), but it's the first one which unblock the PR merge alas.

**Review by dduportal — APPROVED** [review]
Let's try this!

**Comment by lemeurherve** [comment]
@dduportal FTR, I've activated Actions on this repo (the highlighted text is the initial option):
image

**Pull request merged**